### PR TITLE
Add release validity checks

### PR DESCRIPTION
**Pull request by stevemessick**
@devoncarew @pq 

Implement the release-mode validity checks. Add --cwd to allow testing. Add tests.

Most travis builds passed. One failed with:
```bash
HTTP error 500: Internal Server Error at https://pub.dartlang.org/api/packages/collection.
This is likely a transient error. Please try again later.
```

### CODE REVIEW

File: tool/plugin/lib/plugin.dart [hunk 1]
```diff
@@ -8,6 +8,7 @@ import 'dart:io';
 import 'package:args/args.dart';
 import 'package:args/command_runner.dart';
 import 'package:path/path.dart' as p;
+import 'package:git/git.dart';
```
Comment (pq) [inline tool/plugin/lib/plugin.dart]: Sort?

File: tool/plugin/pubspec.yaml [hunk 2]
```diff
@@ -8,6 +8,7 @@ environment:
 dependencies:
   args: "^0.13.7"
   path: "^1.4.2"
+  git: "^0.4.4+2"
```
Comment (pq) [inline tool/plugin/pubspec.yaml]: Sort?

File: tool/plugin/lib/plugin.dart [hunk 3]
```diff
@@ -160,7 +177,7 @@ class BuildCommand extends ProductCommand {
 
   Future<int> doit() async {
```
Comment (pq) [inline tool/plugin/lib/plugin.dart]: General question: do we need to be async?  What bits will take advantage of running asynchronously?  (Sorry if this is obvious!)

**Comment by pq** [comment]
FWIW: I restarted the build and all is green now 👍

File: tool/plugin/lib/plugin.dart (hunk 3 above)
Comment (stevemessick) [reply tool/plugin/lib/plugin.dart]: These methods run a lot of shell commands. Like curl, javac, jar, cp, git, etc. We need to use async to keep it all in order.

**Comment by stevemessick** [comment]
Oops, forgot to run the formatter. I'll do that for the next PR. Thanks.

**Pull request merged**